packages/cli-repl/test/test-shell.ts

```diff
@@ -7,6 +7,8 @@ import { eventually } from './helpers';
 
 
 type SignalType = ChildProcess extends { kill: (signal: infer T) => any } ? T : never;
+type ShellResultState = 'prompt' | 'exit';
+type ShellResult = { state: ShellResultState; exitCode?: number | undefined };
```

Comment:
Btw, we may want to pick another name here to avoid confusion with the "other" ShellResult: https://github.com/mongodb-js/mongosh/blob/c6eacd46ddcb9d0a3ad83754b667ba257f0bdc64/packages/shell-api/src/decorators.ts#L37-L53

Reply:
Thanks, didn't see that. I've renamed it to `TestShellStartupResult`.